src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -92,14 +92,49 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                 // Send request body, if any
                 if (_request.Content != null)
                 {
-                    using (Http2WriteStream writeStream = new Http2WriteStream(this))
+                    try
                     {
-                        await _request.Content.CopyToAsync(writeStream, null, cancellationToken).ConfigureAwait(false);
+                        using (Http2WriteStream writeStream = new Http2WriteStream(this))
+                        {
+                            await _request.Content.CopyToAsync(writeStream, null, cancellationToken).ConfigureAwait(false);
+                        }
+
+                        // Don't wait for completion, which could happen asynchronously.
+                        Task ignored = _connection.SendEndStreamAsync(_streamId);
+                    }
+                    catch (Exception e) when (e is ObjectDisposedException || e.InnerException is ObjectDisposedException)
```

Comment:
What's causing us to get ObjectDisposedExceptions?  It'd be better to avoid them from being thrown in the first place.

Reply:
This is from WriteAsync in Http2WriteStream. When we get error code, we abandon and reset stream if we still sending request. I can add better comment or create distinct exception if that make more sense.